runtime/src/iree/hal/utils/debug_allocator.h

```diff
@@ -0,0 +1,33 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_HAL_UTILS_DEBUG_ALLOCATOR_H_
+#define IREE_HAL_UTILS_DEBUG_ALLOCATOR_H_
+
+#include "iree/base/api.h"
+#include "iree/hal/api.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+// A HAL buffer allocator scribbles debug patterns into all allocated
+// buffers. This can be used to identify incorrect use of memory. Note that this
+// has severe performance implications and should only be used when diagnosing
+// memory correctness issues.
+typedef struct iree_hal_debug_allocator_t iree_hal_debug_allocator_t;
```

Comment:
grammar
```suggestion
// A HAL buffer allocator which scribbles debug patterns into all allocated
// buffers. This can be used to identify incorrect use of memory. Note that this
// has severe performance implications and should only be used when diagnosing
// memory correctness issues.
typedef struct iree_hal_debug_allocator_t iree_hal_debug_allocator_t;
```